nilearn/_utils/data_gen.py

```diff
@@ -330,51 +424,159 @@ def generate_fake_fmri(shape=(10, 11, 12), length=17, kind="noise",
-                                   size=rk,
-                                   replace=False)
-        design_matrices.append(pd.DataFrame(np.random.randn(shape[3], rk),
-                                            columns=columns))
-    mask = Nifti1Image((np.random.rand(*shape[:3]) > .5).astype(np.int8),
+        columns = rand_gen.choice(list(string.ascii_lowercase),
+                                  size=rk,
+                                  replace=False)
+        design_matrices.append(
+            pd.DataFrame(rand_gen.randn(shape[3], rk), columns=columns))
+    mask = Nifti1Image((rand_gen.rand(*shape[:3]) > .5).astype(np.int8),
                        affine)
     return mask, fmri_data, design_matrices
 
 
-def write_fake_fmri_data_and_design(shapes, rk=3, affine=np.eye(4)):
+def write_fake_fmri_data_and_design(shapes,
+                                    rk=3,
+                                    affine=np.eye(4),
+                                    random_state=0):
+    """
+    #TODO help with docstring
+
+    Parameters
+    ----------
+    shapes : List
+        List of shapes in tuple format
+
+    rk : int, optional
+        #TODO what is rk.
+        Default=3.
+
+    affine : numpy.ndarray, optional
+        Affine of returned images.
+        Default=np.eye(4).
+
+    random_state : int or numpy.random.RandomState instance, optional
+        Random number generator, or seed.
+        Default=0.
+
+    Returns
+    -------
+    mask_file : str
+        path to mask file.
+
+    fmri_files : List
+        list of str.
+
+    design_files : List
+        list of str.
+
+    """
     mask_file, fmri_files, design_files = 'mask.nii', [], []
+    rand_gen = check_random_state(random_state)
     for i, shape in enumerate(shapes):
         fmri_files.append('fmri_run%d.nii' % i)
-        data = np.random.randn(*shape)
+        data = rand_gen.randn(*shape)
         data[1:-1, 1:-1, 1:-1] += 100
         Nifti1Image(data, affine).to_filename(fmri_files[-1])
         design_files.append('dmtx_%d.csv' % i)
-        pd.DataFrame(np.random.randn(shape[3], rk),
+        pd.DataFrame(rand_gen.randn(shape[3], rk),
                      columns=['', '', '']).to_csv(design_files[-1])
-    Nifti1Image((np.random.rand(*shape[:3]) > .5).astype(np.int8),
+    Nifti1Image((rand_gen.rand(*shape[:3]) > .5).astype(np.int8),
                 affine).to_filename(mask_file)
     return mask_file, fmri_files, design_files
 
 
-def write_fake_bold_img(file_path, shape, rk=3, affine=np.eye(4)):
-    data = np.random.randn(*shape)
+def write_fake_bold_img(file_path,
+                        shape,
+                        rk=3,
+                        affine=np.eye(4),
+                        random_state=0):
+    """
+    #TODO help with docstring
+
+    Parameters
+    ----------
+    file_path : str
+        output file path.
+
+    shape : tuple
+        Shape of output array.
+
+    rk : int, optional
+        #TODO what is rk.
```

Comment:
It's a good question indeed... 
this parameter isn't used in this function. I suppose it got copy-pasted at some point. 
I suggest we remove it.